lib/tools/adb-emu-commands.js

```diff
@@ -18,29 +18,37 @@ emuMethods.isEmulatorConnected = async function (udid) {
   return false;
 };
 
+emuMethods.checkEmulatorConnected = async function (udid) {
+  if (!_.isUndefined(udid) && !(await this.isEmulatorConnected(udid)) ) {
+    log.errorAndThrow(`Device '${udid}' is not available.`);
+  } else if ((await this.getConnectedEmulators()).length === 0) {
+    log.errorAndThrow('No devices connected');
+  }
+  if (!_.isUndefined(udid)) {
+    await this.setDeviceId(udid);
+  }
+};
+
 emuMethods.fingerprint = async function (fingerprintId, udid = undefined) {
   if (!fingerprintId) {
     log.errorAndThrow('Fingerprint id parameter must be defined');
   }
-
   // the method used only works for API level 23 and above
   let level = await this.getApiLevel();
   if (parseInt(level, 10) < 23) {
     log.errorAndThrow(`Device API Level must be >= 23. Current Api level '${level}'`);
   }
-
-  if (!_.isUndefined(udid) && !(await this.isEmulatorConnected(udid)) ) {
-    log.errorAndThrow(`Device '${udid}' is not available.`);
-  } else if ((await this.getConnectedEmulators()).length === 0) {
-    log.errorAndThrow('No devices connected');
-  }
-
-  if (udid) {
-    await this.setDeviceId(udid);
-  }
+  await this.checkEmulatorConnected(udid);
+  this.resetTelnetAuthToken();
```

Comment:
looks like `resetTelnetAuthToken` needs some `await` love here and below as @mykola-mokhnach pointed out.